pkg/pbq/pbqmanager.go

```diff
@@ -0,0 +1,159 @@
+package pbq
+
+import (
+	"context"
+	"errors"
+	dfv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	"github.com/numaproj/numaflow/pkg/pbq/store"
+	"github.com/numaproj/numaflow/pkg/pbq/store/memory"
+	"github.com/numaproj/numaflow/pkg/pbq/util"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sync"
+)
+
+var NonExistentPBQErr error = errors.New("missing PBQ for the partition")
+
+// Manager helps in managing the lifecycle of PBQ instances
+type Manager struct {
+	options *store.Options
+	pbqMap  map[string]*PBQ
+	log     *zap.SugaredLogger
+}
+
+var createManagerOnce sync.Once
+var pbqManager *Manager
+var createManagerErr error
+
+// NewManager returns new instance of manager
+// We don't intend this to be called by multiple routines.
+func NewManager(ctx context.Context, opts ...store.SetOption) (*Manager, error) {
+	options := store.DefaultOptions()
+	createManagerOnce.Do(func() {
+		for _, opt := range opts {
+			if opt != nil {
+				if err := opt(options); err != nil {
+					createManagerErr = err
+					return
+				}
+			}
+		}
+		pbqManager = &Manager{
+			pbqMap:  make(map[string]*PBQ),
+			options: options,
+			log:     logging.FromContext(ctx),
+		}
+	})
+	return pbqManager, createManagerErr
+}
+
+// ListPartitions returns all the pbq instances
+func (m *Manager) ListPartitions() []*PBQ {
+
+	pbqList := make([]*PBQ, len(m.pbqMap))
+	i := 0
+	for _, val := range m.pbqMap {
+		pbqList[i] = val
+		i++
+	}
+	return pbqList
+}
+
+// GetPBQ returns pbq for the given partitionID, if createIfMissing is set to true
+// it creates and return a new pbq instance
+func (m *Manager) GetPBQ(ctx context.Context, partitionID string, createIfMissing bool, storeType string) (*PBQ, bool, error) {
```

Comment:
also why do we need a storeType for every PQB, if you see startup we imagine it is only just 1 type of store for all the PBQs.